src/grid/GridLoader.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { GridContainerProps, GridContainer } from './GridContainer';
+import { Loading } from '../Loading';
+import { Error } from '../Error';
+
+interface GridLoaderProps {
+  loading: boolean;
+  grid?: GridContainerProps;
+}
+
+const GridLoader: React.FunctionComponent<GridLoaderProps> = (
+  props: GridLoaderProps,
+) => {
+  const { loading, grid } = props;
+
+  if (loading) {
+    return <Loading />;
+  }
+
+  if (grid) {
+    return <GridContainer {...grid} />;
+  }
+
+  return <Error message="We can't load that grid&nbsp;&nbsp;💔" />;
+};
+
+export { GridLoader };
```

Comment:
I think `GridLoader` & `GridContainer` can be combined, their concerns are closely related.

Reply:
They are related but I like the separation because it means that `GridContainer` is entirely focused on the logic of the voting itself. For this reason I'd like to keep them separate - I take the point about whether the creation logic should live in the loader or create container though.